src/WebJobs.Script.WebHost/Helpers/TableStorageHelpers.cs

```diff
@@ -18,27 +20,24 @@ internal static string GetRowKey(DateTime now)
             return string.Format("{0:D19}-{1}", DateTime.MaxValue.Ticks - now.Ticks, Guid.NewGuid());
         }
 
-        internal static async Task<bool> CreateIfNotExistsAsync(CloudTable table, int tableCreationRetries, int retryDelayMS = 1000)
+        internal static async Task<bool> CreateIfNotExistsAsync(TableClient table, int tableCreationRetries, int retryDelayMS = 1000)
         {
             int attempt = 0;
             do
             {
                 try
                 {
-                    if (!table.Exists())
-                    {
-                        return await table.CreateIfNotExistsAsync();
-                    }
+                    return (await table.CreateIfNotExistsAsync()).Value is not null;
```

Comment:
You need to keep that check for `!table.Exists()` -- without it you can end up with tons of AppInsights logs with errors (and thus issues/CRIs). We added this check to prevent those. https://github.com/Azure/azure-functions-host/pull/9555

Reply:
In both cases, 12.8 and 12.6, the conflict that comes from the service and the dependency with a 409 is emitted.  The difference is that in 12.8 is emitted as _informational_ versus 12.6 where is emitted as _warning_ (failed scope):

**12.6:**
```
[Warning] Azure-Core: Error response [955c902a-a10e-44fd-8d53-7c1490ab561f] 409 Conflict (00.5s)
Cache-Control:no-cache
Transfer-Encoding:chunked
Server:Windows-Azure-Table/1.0,Microsoft-HTTPAPI/2.0
x-ms-request-id:783f9c24-e002-001a-0fb2-b1b240000000
x-ms-client-request-id:955c902a-a10e-44fd-8d53-7c1490ab561f
x-ms-version:REDACTED
X-Content-Type-Options:REDACTED
Date:Wed, 29 May 2024 10:28:38 GMT
Content-Type:application/json; odata=minimalmetadata; streaming=true; charset=utf-8
```
**12.8:**
```
[Informational] Azure-Core: Response [d0228e9f-9e96-481a-b2eb-513c9764210d] 409 Conflict (00.4s)
Cache-Control:no-cache
Transfer-Encoding:chunked
Server:Windows-Azure-Table/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id:b51963d4-3002-0031-2bb5-b1c6f8000000
x-ms-client-request-id:d0228e9f-9e96-481a-b2eb-513c9764210d
x-ms-version:REDACTED
X-Content-Type-Options:REDACTED
Preference-Applied:REDACTED
Date:Wed, 29 May 2024 10:43:36 GMT
Content-Type:application/json;odata=minimalmetadata;streaming=true;charset=utf-8
```
In App Insights translates to a success call (even if 409) in 12.8 versus a failed dependency call in 12.6.

**12.6:**

![image](https://github.com/Azure/azure-functions-host/assets/1939768/458e0e03-afef-4315-87ea-0e3fb4638d0f)

**12.8:**

![image](https://github.com/Azure/azure-functions-host/assets/1939768/ce789a18-c97c-41ad-9b1c-90e4152b00af)

I'm OK with both options: checking if exists (adding an extra successful dependency call to list the tables that should end in 200) or letting the 409 as a successful call. That said, with 12.8 it doesn't seem to appear as a warning so I think we should be fine. 

Let me know what you think @brettsam @jviau

Reply:
@cjaliaga how often is this call made? And what is the telemetry item you have shared - dependency or trace? I wonder how this will look if customer (or we) eventually includes this in a metric. There is no separate success/failed attribute from status code, so there wouldn't be a way to represent "409, but expected". In this case it may be better to make the GET call.

Reply:
Every time we flush the logs so it might be fairly frequent. The telemetry item was a trace and the screenshot shows a dependency item as well. I agree with your reasoning, let's check if it already exists.